test/replication-luatest/bootstrap_strategy_test.lua

```diff
@@ -179,3 +181,249 @@ g_auto.test_sync_waits_for_all_connected = function(cg)
+            replication_timeout = 0.1,
+            bootstrap_strategy = 'config',
+            bootstrap_leader = nil,
+            replication = server.build_listen_uri('server1'),
+        },
+        env = {
+            ['TARANTOOL_RUN_BEFORE_BOX_CFG'] = set_log_before_cfg,
+        },
+    }
+end)
+
+g_config.test_no_leader = function(cg)
+    cg.replica_set:start{wait_until_ready = false}
+    local logfile = fio.pathjoin(cg.server1.workdir, 'server1.log')
+    local empty_leader_msg = "the option can't be empty when bootstrap " ..
+                             "strategy is 'config'"
+    t.helpers.retrying({}, function()
+        t.assert(server:grep_log(empty_leader_msg, nil, {filename = logfile}))
+    end)
+end
+
+g_config.before_test('test_single_leader', function(cg)
+    cg.replica_set = replica_set:new{}
+    cg.server1 = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = {
+            replication_timeout = 0.1,
+            bootstrap_strategy = 'config',
+            bootstrap_leader = server.build_listen_uri('server1'),
+            replication = server.build_listen_uri('server1'),
+        },
+    }
+end)
+
+g_config.test_single_leader = function(cg)
+    cg.replica_set:start()
+    cg.server1:exec(function()
+        local t = require('luatest')
+        t.assert_equals(box.info.status, 'running', 'server is working')
+    end)
+end
+
+g_config.after_test('test_single_leader', function(cg)
+    cg.server1:stop()
+end)
+
+local g_config_success = t.group('gh-7999-bootstrap-strategy-config-success', {
+                       {leader = 'server3'},
+                       {leader = uuid3},
+                   })
+
+g_config_success.before_each(function(cg)
+    cg.leader = cg.params.leader
+    -- cg.params can't have "/" for some reason, so recreate the path here.
+    if string.match(cg.leader, 'server3') then
+        cg.leader = server.build_listen_uri(cg.leader)
+    end
+end)
+
+g_config_success.after_each(function(cg)
+    cg.replica_set:drop()
+end)
+
+g_config_success.before_test('test_correct_bootstrap_leader', function(cg)
+    cg.replica_set = replica_set:new{}
+    cg.server1 = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = {
+            bootstrap_strategy = 'config',
+            bootstrap_leader = cg.leader,
+            instance_uuid = uuid1,
+            replication = {
+                server.build_listen_uri('server1'),
+                server.build_listen_uri('server2'),
+                server.build_listen_uri('server3'),
+            },
+            replication_timeout = 0.1,
+        },
+    }
+    cg.server2 = cg.replica_set:build_and_add_server{
+        alias = 'server2',
+        box_cfg = {
+            replicaset_uuid = uuida,
+            instance_uuid = uuid2,
+        }
+    }
+    cg.server3 = cg.replica_set:build_and_add_server{
+        alias = 'server3',
+        box_cfg = {
+            replicaset_uuid = uuidb,
```

Comment:
Why are there 2 different replicaset UUIDs in replicas of the same `cg.replica_set`?